File: app/src/main/java/com/ladsoft/bakingapp/activity/di/RecipeActivityModule.java
package com.ladsoft.bakingapp.activity.di;

import com.ladsoft.bakingapp.activity.RecipeActivity;
import com.ladsoft.bakingapp.mvp.RecipeMvp;
import com.ladsoft.bakingapp.mvp.presenter.RecipePresenter;
import com.ladsoft.bakingapp.mvp.presenter.StepPresenter;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class RecipeActivityModule {
    @Binds
    abstract RecipeMvp.View providesView(RecipeActivity view);

    @Provides
    static RecipePresenter providesPresenter(RecipeMvp.Model model){ return new RecipePresenter(model); }

    @Provides
    static StepPresenter providesStepPresenter() { return new StepPresenter(); }
}

File: app/src/main/java/com/ladsoft/bakingapp/activity/di/RecipeActivityModule.kt
package com.ladsoft.bakingapp.activity.di

import com.ladsoft.bakingapp.activity.RecipeActivity
import com.ladsoft.bakingapp.mvp.RecipeMvp
import com.ladsoft.bakingapp.mvp.presenter.RecipePresenter
import dagger.Module
import dagger.Provides

@Module
class RecipeActivityModule {
    @Provides
    fun providesView(view: RecipeActivity): RecipeMvp.View = view

    @Provides
    fun providesPresenter(model: RecipeMvp.Model): RecipePresenter = RecipePresenter(model)
}